panos_upgrade_assurance/firewall_proxy.py

```diff
@@ -455,14 +428,18 @@ def get_support_license(self) -> dict:
         - the support level.
         """
         result = {}
-        response = self.op_parser(cmd="request support check", return_xml=True)
-
-        result["support_expiry_date"] = response.findtext(
-            "./SupportInfoResponse/Support/ExpiryDate"
-        )
-        result["support_level"] = response.findtext(
-            "./SupportInfoResponse/Support/SupportLevel"
-        )
+        try:
+            response = self.op_parser(cmd="request support check", return_xml=True)
+        except PanXapiError as exp:
+            # This is an ugly hack to narrow down the cause of the exception to update server connectivity issues.
+            # Unfortunately the exception class is generic in this case. What is easy to identify is the message.
+            if str(exp) == "Failed to check support info due to Unknown error. Please check network connectivity and try again.":
```

Comment:
Did you find the source of these exceptions messages, I saw a couple of similar messages on the internet for dynamic updates which has the "network connectivity" phrase in the exception message as well. Maybe instead of checking for exact equality we may check if the message includes "network connectivity" which may catch similar exceptions. I'm not sure what would be better though.

```python
if "network connectivity" in str(exp)
```

Reply:
I did not find a source for this, I just broke the connectivity between the update server and the device and I got this error message. Worst case scenario, we will still raise an exception with the `raise exp` in `else`. We just won't catch it in the `CheckFirewall` class.